### ISSUE TITLE
Exposing call origin in ink!

### ISSUE THREAD

**Issue by coreggon11** [paritytech/ink#1792]
In ink! we can tell if a caller is the origin pretty easily by calling the `Env::caller_is_origin` method. However, sometimes the users want to know who the origin is.

Are there any plans on adding this functionality / expose the call origin in ink!? Or are there specific reasons why this is not contained in ink!?

**Comment by h4x3rotab** [comment]
Yes, it's really confusing why it's not exposed. We have a similar issue [here](https://github.com/Phala-Network/phala-blockchain/pull/1478#pullrequestreview-1779381239). We have implemented our own version as a Chain Extension in Phat Contract.
